pkg/scheduler/scheduler_test.go

```diff
@@ -994,14 +995,16 @@ func TestFrameworkHandler_IterateOverWaitingPods(t *testing.T) {
 			fakeClient := fake.NewSimpleClientset(objs...)
 			informerFactory := informers.NewSharedInformerFactory(fakeClient, 0)
 			eventBroadcaster := events.NewBroadcaster(&events.EventSinkImpl{Interface: fakeClient.EventsV1()})
+			defer eventBroadcaster.Shutdown()
 			eventRecorder := eventBroadcaster.NewRecorder(scheme.Scheme, fakePermit)
 
 			outOfTreeRegistry := frameworkruntime.Registry{
 				fakePermit: newFakePermitPlugin(eventRecorder),
 			}
 
 			_, ctx := ktesting.NewTestContext(t)
-			ctx, cancel := context.WithCancel(ctx)
+			// timeout equals to the permit plugin waiting time.
+			ctx, cancel := context.WithTimeout(ctx, 100*time.Second)
```

Comment:
Where is that time specified? What if it changes?

Reply:
what is the longest we would expect this to take to settle to the expected result? I would hope a few seconds, not a minute and a half

Reply:
Generally permit plugin will wait for a relative long time like 100s but looking into the codes, only used in one case and no waiting logic added, 5s seems enough, I tested locally, under race mode, almost 3s. But to be more tolerant, so 1) I set this to 10s and 2) also make it a constant.